examples/usercall-extension/runner/src/main.rs

```diff
@@ -0,0 +1,112 @@
+/* Copyright (c) Fortanix, Inc.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+extern crate aesm_client;
+extern crate enclave_runner;
+extern crate sgxs_loaders;
+
+use aesm_client::AesmClient;
+use enclave_runner::{UsercallExtension, EnclaveBuilder, SyncStream};
+use sgxs_loaders::isgx::Device as IsgxDevice;
+
+use std::io::{ErrorKind as IoErrorKind, Result as IoResult, Write, Read};
+use std::process::{Command, Child, Stdio};
+use std::sync::{Arc, Mutex};
+
+struct CatService {
+   c : Arc<Mutex<Child>>
+}
+impl CatService {
+   fn new() -> CatService {
+     CatService {
+        c : Arc::new(Mutex::new(Command::new("/bin/cat")
+            .stdout(Stdio::piped())
+            .stdin(Stdio::piped())
+            .spawn()
+            .expect("Failed to start cat process")))
+        }
+   }
+}
+impl SyncStream for CatService {
+
+    fn read(&self, buf: &mut [u8]) -> IoResult<usize> {
+        self.c.lock().unwrap().stdout.as_mut().expect("failed to get stdout").read(buf)
+    }
+
+    fn write(&self, buf: &[u8]) -> IoResult<usize> {
+        self.c.lock().unwrap().stdin.as_mut().expect("failed to get stdin").write(buf)
+    }
+
+    fn flush(&self) -> IoResult<()> {
+        self.c.lock().unwrap().stdin.as_mut().expect("failed to get stdin").flush()
+    }
+}
+
+#[derive(Debug)]
+struct ExternalService;
+impl UsercallExtension for ExternalService {
+    fn connect_stream(
+        &self,
+        addr: &[u8],
+        local_addr: Option<&mut String>,
+        peer_addr: Option<&mut String>,
+    ) -> IoResult<Option<Box<SyncStream>>> {
+        let name = String::from_utf8(addr.to_vec()).map_err(|_| IoErrorKind::ConnectionRefused)?;
```

Comment:
Don't error out if the address is not UTF-8

Reply:
In particular, return `Ok(None)` instead.